### PR TITLE
Octopart: Switch from api v2 to v3 (fixes #1595)

### LINKED ISSUE #1595: Octopart: Migrate to API v3 -- API v2 deprecating March 20!
Octopart will be officially turning off Octopart API v2 on **March 20, 2015** and **will not be functional** after March 20. 

Let's update it fast!

Here's the documentation for v3: https://octopart.com/api/docs/v3/overview

Code: https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/Octopart.pm#L18

https://duck.co/ia/view/octopart

### PR DESCRIPTION
**Pull request by killerfish**
fixes #1595 
@moollaza let me know if the switch to v3 slows down spice calls, ill see if it could be optimized.

### CODE REVIEW

File: share/spice/octopart/octopart.js [hunk 1]
```diff
@@ -50,18 +50,18 @@ function ddg_spice_octopart (api_result) {
 			sourceUrl : 'http://octopart.com/partsearch#search/requestData&q=' + api_result.request.q
```
Comment (moollaza) [inline share/spice/octopart/octopart.js]: This is actually currently broken, can you please fix?

Should be: `http://octopart.com/search?q=`

Comment (killerfish) [inline share/spice/octopart/octopart.js]: done

**Comment by jagtalon** [comment]
Nice! Gotta get this in before the 20th!

**Comment by moollaza** [comment]
@killerfish thanks for getting this going! I gave it a try and unfortunately it wasn't working -- seems like some of the params you were using no longer exist.

I noticed the code code be simplified and cleaned up a bit and pushed to a new PR #1651 -- would you be interested in continuing that? I noticed that the detail pane isn't showing so something still needs to be fixed...

**Comment by moollaza** [comment]
Actually i think I manged to just about finish it all up...

**Comment by moollaza** [comment]
@killerfish apologies -- I was actually not using API v3 when testing this which is why it seemed broken. @jagtalon pointed that out to me while testing my PR. I've changed my code to use what you had before and I've added some other updates as well.

**Pull request closed without merging**